lib/ProductOpener/KnowledgePanels.pm

```diff
@@ -838,4 +841,55 @@ sub create_physical_activities_panel($$$) {
 }
 
 
+=head2 create_ingredients_panel ( $product_ref, $target_lc, $target_cc )
+
+Creates a knowledge panels with the list of ingredients.
+
+=head3 Arguments
+
+=head4 product reference $product_ref
+
+Loaded from the MongoDB database, Storable files, or the OFF API.
+
+=head4 language code $target_lc
+
+Returned attributes contain both data and strings intended to be displayed to users.
+This parameter sets the desired language for the user facing strings.
+
+=head4 country code $target_cc
+
+=cut
+
+sub create_ingredients_panel($$$) {
+
+	my $product_ref = shift;
+	my $target_lc = shift;
+	my $target_cc = shift;
+
+	$log->debug("create ingredients panel", { code => $product_ref->{code}, nutriscore_data => $product_ref->{nutriscore_data} }) if $log->is_debug();
+
+	# try to display ingredients in the requested language if available
+
+	my $ingredients_text = $product_ref->{ingredients_text};
+    my $ingredients_text_with_allergens = $product_ref->{ingredients_text_with_allergens};
+	my $ingredients_text_lc = $product_ref->{lc};
+
+	if ((defined $product_ref->{"ingredients_text" . "_" . $target_lc}) and ($product_ref->{"ingredients_text" . "_" . $target_lc} ne '')) {
```

Comment:
(thinking) I find it's a pity not to have some small utils functions for this kind of pattern, that would make it more readable. Something like `is_non_empty_text($product_ref->{"ingredients_text" . "_" . $target_lc})`